src/Microsoft.AspNetCore.Razor.Language/Legacy/DirectiveChunkGenerator.cs

```diff
@@ -17,6 +21,19 @@ public DirectiveChunkGenerator(DirectiveDescriptor descriptor)
 
         public DirectiveDescriptor Descriptor { get; }
 
+        public List<RazorDiagnostic> Diagnostics
```

Comment:
This is an internal API (the chunk generator) which doesn't need to have the extra pieces of the diagnostic collection. We only use the `RazorDiagnosticCollection` to have control over the public API surface of our IR.